components/browser/tabstray/src/main/java/mozilla/components/browser/tabstray/TabsAdapter.kt

```diff
@@ -48,10 +62,32 @@ open class TabsAdapter(
 
     override fun getItemCount() = tabs?.list?.size ?: 0
 
+    var listenerAdded = false
+    var frame = 1
+
     override fun onBindViewHolder(holder: TabViewHolder, position: Int) {
         val tabs = tabs ?: return
 
         holder.bind(tabs.list[position], position == tabs.selectedIndex, this)
+
+        Log.e("lol", "onBind")
+        if (!listenerAdded) {
+            listenerAdded = true
+            holder.itemView.viewTreeObserver.addOnPreDrawListener {
+                if (frame == 4) {
+                    // On the 4th frame with animations enabled, the tabs in the tray start to fade in:
+                    // tested with 1 tab & 8 tabs.
+                    Handler(Looper.getMainLooper()).post {
+                        Timings.tabsTrayTabsEnd = SystemClock.elapsedRealtime()
```

Comment:
The `tabsTrayTabsEnd` sounds like we want to see how long it takes to present all the tabs. This is the bind call that's executed for each tab and not the entire tabs tray. Not sure if that's intentional. 🤔

Reply:
Ah, I see now (5 days later 😅 ) what we're trying to do! `ViewHolder.onViewAttachedToWindow` might be more closer to including the whole time it takes to create, bind & draw, if that's something we want to measure.